### ISSUE TITLE
Bard: Unable to get __Secure-1PSID cookie

### ISSUE THREAD

**Issue by TroublesInParadise** [Josh-XT/Agent-LLM#29]
Looks like it may be related to Google blocking automation with selenium? Just got access to Bard so I'm testing the provider.
Edit: stumbled upon this just now, maybe could be a fix. Will link just in case: https://github.com/ra83205/google-bard-api

127.0.0.1 - - [22/Apr/2023 13:43:08] "GET /api/agent HTTP/1.1" 200 -
127.0.0.1 - - [22/Apr/2023 13:43:08] "GET /api/agent/Rosemore HTTP/1.1" 200 -
127.0.0.1 - - [22/Apr/2023 13:43:08] "GET /api/agent/undefined/command HTTP/1.1" 200 -
127.0.0.1 - - [22/Apr/2023 13:43:08] "GET /api/agent/Rosemore/command HTTP/1.1" 200 -
127.0.0.1 - - [22/Apr/2023 13:43:10] "POST /api/agent/Rosemore/task HTTP/1.1" 500 -
Traceback (most recent call last):
  File "C:\Users\Stephan\AppData\Local\Programs\Python\Python310\lib\site-packages\flask\app.py", line 2551, in __call__
    return self.wsgi_app(environ, start_response)
  File "C:\Users\Stephan\AppData\Local\Programs\Python\Python310\lib\site-packages\flask\app.py", line 2531, in wsgi_app
    response = self.handle_exception(e)
  File "C:\Users\Stephan\AppData\Local\Programs\Python\Python310\lib\site-packages\flask_restful\__init__.py", line 271, in error_router
    return original_handler(e)
  File "C:\Users\Stephan\AppData\Local\Programs\Python\Python310\lib\site-packages\flask_cors\extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "C:\Users\Stephan\AppData\Local\Programs\Python\Python310\lib\site-packages\flask\app.py", line 2528, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Stephan\AppData\Local\Programs\Python\Python310\lib\site-packages\flask\app.py", line 1825, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Stephan\AppData\Local\Programs\Python\Python310\lib\site-packages\flask_restful\__init__.py", line 271, in error_router
    return original_handler(e)
  File "C:\Users\Stephan\AppData\Local\Programs\Python\Python310\lib\site-packages\flask_cors\extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "C:\Users\Stephan\AppData\Local\Programs\Python\Python310\lib\site-packages\flask\app.py", line 1823, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Stephan\AppData\Local\Programs\Python\Python310\lib\site-packages\flask\app.py", line 1799, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "C:\Users\Stephan\AppData\Local\Programs\Python\Python310\lib\site-packages\flask_restful\__init__.py", line 467, in wrapper
    resp = resource(*args, **kwargs)
  File "C:\Users\Stephan\AppData\Local\Programs\Python\Python310\lib\site-packages\flask\views.py", line 107, in view
    return current_app.ensure_sync(self.dispatch_request)(**kwargs)
  File "C:\Users\Stephan\AppData\Local\Programs\Python\Python310\lib\site-packages\flask_restful\__init__.py", line 582, in dispatch_request
    resp = meth(*args, **kwargs)
  File "C:\Users\Stephan\Desktop\Vicuna\Agent-LLM\app.py", line 114, in post
    agent_instances[agent_name] = AgentLLM(agent_name)
  File "C:\Users\Stephan\Desktop\Vicuna\Agent-LLM\AgentLLM.py", line 43, in __init__
    self.ai_instance = ai_module.AIProvider()
  File "C:\Users\Stephan\Desktop\Vicuna\Agent-LLM\provider\bard.py", line 20, in __init__
    raise Exception("Unable to get __Secure-1PSID cookie.")
Exception: Unable to get __Secure-1PSID cookie.

**Comment by TroublesInParadise** [comment]
@Josh-XT Update! Seems to work but I got an error that flies away way too fast and gets hidden away by the massive Bard output. So, Agent UI shows nothing but the console does. It crashed on Google's end after a few runs without reading or writing any file or executing commands:
![image](https://user-images.githubusercontent.com/131233096/233823172-0c430a6c-f579-4d1e-86a2-0e0cd682ea20.png)
This came out after tinkering with Bard.py:

```
from selenium import webdriver
from Bard import Chatbot
from Config import Config

CFG = Config()

class AIProvider:
    def __init__(self):
        if CFG.AI_PROVIDER.lower() == "bard":
            self.chatbot = Chatbot("TOKEN-GOES-HERE")

    def instruct(self, prompt, seed=None):
        response = self.chatbot.ask(prompt)
        print(response)
        return response['content'].replace("\n", "\n")

```

We're getting there maybe hopefully somewhat?

**Comment by Josh-XT** [comment]
Thanks, that was very helpful.  I just updated it.

**Comment by TroublesInParadise** [comment]
Hey, I managed to snag the error:
  File "C:\Users\Stephan\AppData\Local\Programs\Python\Python310\lib\site-packages\fastapi\routing.py", line 163, in run_endpoint_function
    return await dependant.call(**values)
  File "C:\Users\Stephan\Desktop\Vicuna\Agent-LLM\app.py", line 170, in get_task_output
    return TaskOutput(output=output, message="Task agent is still running")
  File "pydantic\main.py", line 341, in pydantic.main.BaseModel.__init__
pydantic.error_wrappers.ValidationError: 1 validation error for TaskOutput
output
  str type expected (type=type_error.str)
Response: Sure, here is my initial task list:

----
Not sure what it means that the task agent is running, I pulled and ran a new agent with a goal to make a c++ project

Command not recognized: **

* This task seems straightforward, but it will require some careful planning and execution.
* I will need to first read the existing Python files to understand how the project works.
* Once I have a good understanding of the project, I can start converting the code to C++.
* I will need to be careful to make sure that the C++ code is equivalent to the Python code, and that it still works correctly.
* I will also need to make sure that the C++ code is well-written and easy to understand.
